### PR TITLE
Update mkdocs and expand nav by default, to make the nested docs more…

### PR DESCRIPTION
**Pull request by helenb**
[Part of this ticket](https://torchbox.monday.com/boards/1435178786/pulses/1435178923)

### Description of Changes Made

- Update mkdocs and expand the navigation by default, to make the nested docs more obvious
- Various updates to the docs
- Fix broken links

### How to Test

Run `mkdcos serve and visit http://0.0.0.0:8001/
You may need to run `fab stop` and `fab start` for docker to pick up the changes to the yaml file

### Screenshots

<details>
  <summary>Expand to see more</summary>

FE docs more obvious when you load the docs:

Screenshot 2024-04-19 at 12 59 14

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [x] Updated build docs
- [ ] Updated editor guidelines (https://docs.google.com/document/d/1PAWccdQ4tfaZsrEWmpDhvP3GH5RRmBOARFVp4b-kje8/edit?usp=sharing)
- [ ] Not required

#### Browser testing

- [ ] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [x] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] HTML validation passes
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [x] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [x] Changes are not relevant the pattern library

### CODE REVIEW

File: docs/front-end/themes_and_modes.md [hunk 1]
```diff
@@ -10,6 +10,12 @@ An editor has an option to select a theme on each page. Selecting a theme on a p
 
 There are currently 3 themes in use: coral, lagoon and banana. The CSS to set the theme is an html class of either `.theme-coral`, `.theme-lagoon` or `.theme-banana`.
 
+???+ note
+
+    If you are adding a new theme, check the colour contrast for all the new accent colours added, used in the drop-caps (remember to check both dark mode and light mode). They need to pass colour contrast as if the entire drop-cap was filled with that colour.
+
+    The drop-cap svgs blend two of the accent colours, and the contrast of the blended colour also needs to be checked. The [colour contrast checker](https://chromewebstore.google.com/detail/colour-contrast-checker/nmmjeclfkgjdomacpcflgdkgpphpmnfe?hl=en-GB&utm_source=ext_sidebar) is a useful chrome extension to assist with this.
```
Comment (helenb) [inline docs/front-end/themes_and_modes.md:17]: @chris-lawton can you remember if I have described the drop caps issue correctly here?

Comment (chris-lawton) [reply docs/front-end/themes_and_modes.md:17]: The initial issue with them was that they used mix-blend-mode which produced colours that didn't pass. When we switched that to opacity instead they were ok. I think what you've written here is fine - the key is to make sure that all the colours are checked against light/dark backgrounds.

**Review by chris-lawton — APPROVED** [review]
This all looks good to me, thanks @helenb

File: docs/front-end/themes_and_modes.md (hunk 1 above)
Comment (helenb) [reply docs/front-end/themes_and_modes.md:17]: Would it be more accurate to say "The drop-caps svgs use a semi-transparent version of the accent colours, and the contrast of the resulting colour also needs to be checked".

Comment (chris-lawton) [reply docs/front-end/themes_and_modes.md:17]: Yep, sounds good 👍

**Pull request merged**